pygeos/geometry.py

```diff
@@ -537,3 +539,86 @@ def get_num_geometries(geometry):
     0
     """
     return lib.get_num_geometries(geometry)
+
+
+@requires_geos("3.6.0")
+@multithreading_enabled
+def get_precision(geometry):
+    """Get the precision of the geometry.
+
+    If the precision has not been previously set, it will be 0 (floating point precision).
+    Otherwise, it will return the grid size that was set on the geometry.
+
+    Returns NaN for not-a-geometry values.
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+
+    See also
+    --------
+    set_precision
+
+    Examples
+    --------
+    >>> get_precision(Geometry("POINT (1 1)")) == 0
+    True
+    >>> geometry = set_precision(Geometry("POINT (1 1)"), 1)
+    >>> get_precision(geometry) == 1
+    True
+    >>> np.isnan(get_precision(None))
+    True
+    """
+    return lib.get_precision(geometry)
+
+
+@requires_geos("3.6.0")
+@multithreading_enabled
+def set_precision(geometry, grid_size=0, preserve_topology=False):
+    """Returns geometry with the precision set to the precision grid size, rounding all
```

Comment:
This line is too long, the 'short description' should fit on 1 line.

Also I'd like to keep the option open for plural goemetry input. So instead of "Set the precision of the Geometry" I would like "Set the precision of a Geometry".
- I didn't look at the tests, in general I try to stick with the 1-assert-per-unittest rule, of course a few are not a problem, but these have very many... are some print functions in the tests.